status-im/status-react
# Dapps Typed Data request improvements

fixes #21278

### Summary

[User story](https://www.notion.so/I-would-like-to-understand-the-contents-of-typed-data-signing-requests-e36ddbe7df5f4afb98f17e0d25b3e442?pvs=4)

Improved the UI of the wallet connect typed data request by flattening the data and displaying it to the user, as seen in the screenshot below.

![image](https://github.com/user-attachments/assets/edabfff7-e27e-4b81-96ad-665c9640f6b4)

### Steps to test


- Open Status
- Connect to dapp (e.g. https://react-app.walletconnect.com/ will work on mainnet only)
- Generate a `eth_signTypedData_v4` request
- See the request data

status: ready
